util/qc_executor.py

```diff
@@ -98,48 +90,65 @@ def qc_check(self, parameter, dataset):
             if 'strict_validation' not in qcs[function_name]:
                 local_qc_args[function_name]['strict_validation'] = False
 
-            qc_process = None
-            try:
-                # call qc function in a separate process to deal with crashes, e.g. segfaults
-                module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name)
-                                                 .first().owner)
-                parent_conn, child_conn = Pipe()
-                qc_process = Process(target=self.qc_process_wrapper, args=(child_conn, getattr(module, function_name),
-                                                                           local_qc_args.get(function_name)))
-                qc_process.start()
-                # close unused end of duplex pipe
-                child_conn.close()
-                results = parent_conn.recv()
-                qc_process.join()
-
-                # if qc function threw an exception, log it and try the next function
-                if isinstance(results, Exception):
-                    log.error('<%s> Failed to execute QC %s \n%s', self.request_id, function_name, results)
+            module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name)
+                                             .first().owner)
+            # call qc function in a separate process to deal with crashes, e.g. segfaults
+            r, w = os.pipe()
+            processid = os.fork()
+            if processid == 0:
+                # child process
+                try:
+                    os.close(r)
+                    w = os.fdopen(w, 'w')
+                    # run the qc function
+                    results = getattr(module, function_name)(**local_qc_args.get(function_name))
+                    # convert the np.ndarray into a string for sending over pipe
+                    bytes = io.BytesIO()
+                    np.savez(bytes, x=results)
+                    results_string = bytes.getvalue()
+                    w.write(results_string)
+                except (TypeError, ValueError) as e:
+                    log.exception('<%s> Failed to execute QC %s %r', self.request_id, function_name, e)
+                    w.write(EXCEPTION_MESSAGE)
+                finally:
+                    w.close()
+                    os._exit(0)
+            else:
+                # parent process
+                os.close(w)
+                r = os.fdopen(r)
+                results_string = r.read()
+                # check for failure to produce results
+                if not results_string or results_string == "":
```

Comment:
redundant, just use:

`if not results_string:`

Reply:
Done